### ISSUE TITLE
val_dataloader participate in training?

### ISSUE THREAD

**Issue by wunianjian** [PyTorchLightning/pytorch-lightning#1928]
## ❓ Questions and Help

It's very strange. I just use
```
trainer.fit(net, train_dataloader=data.train_loader, val_dataloaders=data.val_loader)
trainer.test(test_dataloaders=data.test_loader)
```
to train, validate and test.
I am sure train_loader, val_loader and test_loader are from independent dataset.
But the test result turns out to be much worse than val result, and val result is close to train result, which seems that the val set takes part in training. I can't figure out what happened. Is this a bug or my way is wrong?

**Comment by MaharshiYeluri01** [comment]
What's the metric you are using to compare the performance train,val and test?

**Comment by wunianjian** [comment]
RMSE. I am working on a review score prediction task.

**Comment by awaelchli** [comment]
If I don't read it wrongly, there is nothing surprising about that.
If the datasets are all different, it is certainly possible or even reasonable to expect in general that the test error will be higher than the training and validation error.
Obvious reasons:
- overfitting
- early stopping
the combination of the two points could be called "overfitting to validation set", and so even if your test set is close to the validation set (in terms of statistics) it can yield a high error.

**Comment by awaelchli** [comment]
If there is a bug in your code, this could help:
- try the validation dataset in the test loop, it should yield the same metric as in validation loop
- double check that you average your metric accross the whole dataset
